okhttp/src/main/java/okhttp3/internal/tls/OkHostnameVerifier.kt

```diff
@@ -14,130 +14,79 @@
  *  See the License for the specific language governing permissions and
  *  limitations under the License.
  */
-package okhttp3.internal.tls;
-
-import java.security.cert.Certificate;
-import java.security.cert.CertificateParsingException;
-import java.security.cert.X509Certificate;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.List;
-import java.util.Locale;
-import javax.net.ssl.HostnameVerifier;
-import javax.net.ssl.SSLException;
-import javax.net.ssl.SSLSession;
-
-import static okhttp3.internal.Util.verifyAsIpAddress;
+package okhttp3.internal.tls
+
+import okhttp3.internal.Util.verifyAsIpAddress
+import java.security.cert.CertificateParsingException
+import java.security.cert.X509Certificate
+import java.util.Locale
+import javax.net.ssl.HostnameVerifier
+import javax.net.ssl.SSLException
+import javax.net.ssl.SSLSession
 
 /**
- * A HostnameVerifier consistent with <a href="http://www.ietf.org/rfc/rfc2818.txt">RFC 2818</a>.
+ * A HostnameVerifier consistent with [RFC 2818][rfc_2818].
+ *
+ * [rfc_2818]: http://www.ietf.org/rfc/rfc2818.txt
  */
-public final class OkHostnameVerifier implements HostnameVerifier {
-  public static final OkHostnameVerifier INSTANCE = new OkHostnameVerifier();
-
-  private static final int ALT_DNS_NAME = 2;
-  private static final int ALT_IPA_NAME = 7;
-
-  private OkHostnameVerifier() {
-  }
-
-  @Override
-  public boolean verify(String host, SSLSession session) {
-    try {
-      Certificate[] certificates = session.getPeerCertificates();
-      return verify(host, (X509Certificate) certificates[0]);
-    } catch (SSLException e) {
-      return false;
+object OkHostnameVerifier : HostnameVerifier {
+  private const val ALT_DNS_NAME = 2
+  private const val ALT_IPA_NAME = 7
+
+  override fun verify(host: String, session: SSLSession): Boolean {
+    return try {
+      verify(host, session.peerCertificates[0] as X509Certificate)
```

Comment:
tioli: first looks nicer even for arrays

Reply:
I think leave it, ’cause it adds an unnecessary range check and replaces an instruction with a method call